File: 2025-2/sesion-de-clase-25/java/src/main/java/Retiro.java
/**
 *
 * @author eric
 */
public class Retiro implements Runnable {
    private final double monto;
    private final ICuentaBancaria cuenta;
    
    public Retiro(ICuentaBancaria cuenta, double monto) {
        this.monto = monto;
        this.cuenta = cuenta;
    }
    
    @Override
    public void run() {
        String nombreCliente = Thread.currentThread().getName();
        this.cuenta.retirar(nombreCliente, monto);
    }
}

File: 2026-1/sesion-de-clase-24/net/sesion-de-clase-24/Ejercicio4/Retiro.cs
using System.Threading;

namespace Ejercicio4;

public class Retiro
{
    public CuentaBancaria Cuenta { get; set; }

    public double Monto { get; set; }

    public Retiro(CuentaBancaria cuenta, double monto)
    {
        Cuenta = cuenta;
        Monto = monto;
    }

    public void Run()
    {
        string nombreCliente = Thread.CurrentThread.Name ?? "Sin nombre";
        Cuenta.Retirar(nombreCliente, Monto);
    }
}
